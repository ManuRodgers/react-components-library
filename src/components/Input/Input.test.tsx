import {
  RenderResult,
  cleanup,
  fireEvent,
  render,
} from '@testing-library/react';
import React from 'react';

import Input, { InputProps } from './Input';

const defaultProps: InputProps = {
  onChange: jest.fn(),
  placeholder: 'test-input',
};
function renderInput(props: Partial<InputProps> = {}): RenderResult {
  return render(<Input {...defaultProps} {...props} />);
}
let wrapper: RenderResult;
describe('test Input component', () => {
  afterEach(cleanup);
  it('should render the correct default Input', async () => {
    wrapper = renderInput();
    const testNode = (await wrapper.findByPlaceholderText(
      'test-input'
    )) as HTMLInputElement;
    expect(testNode).toBeInTheDocument();
    expect(testNode).toHaveClass('viking-input-inner');
    fireEvent.change(testNode, { target: { value: '23' } });
    expect(defaultProps.onChange).toHaveBeenCalled();
    expect(testNode.value).toEqual('23');
  });

  it('should render the disabled Input on disabled property', async () => {
    wrapper = renderInput({ disabled: true, placeholder: 'disabled' });
    const testNode = (await wrapper.findByPlaceholderText(
      'disabled'
    )) as HTMLInputElement;
    expect(testNode.disabled).toBeTruthy();
  });

  it('should render different input sizes on size property', () => {
    wrapper = renderInput({ size: 'lg', placeholder: 'sizes' });
    const testContainer = wrapper.container.querySelector(
      '.viking-input-wrapper'
    );
    expect(testContainer).toHaveClass('input-size-lg');
  });
  it('should render prepend and append element on prepend/append property', () => {
    const { queryByText, container } = render(
      <Input placeholder="pend" prepend="https://" append=".com" />
    );
    const testContainer = container.querySelector('.viking-input-wrapper');
    expect(testContainer).toHaveClass(
      'input-group input-group-append input-group-prepend'
    );
    expect(queryByText('https://')).toBeInTheDocument();
    expect(queryByText('.com')).toBeInTheDocument();
  });
  it('matches snapshot', async () => {
    const { asFragment } = render(<Input {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
