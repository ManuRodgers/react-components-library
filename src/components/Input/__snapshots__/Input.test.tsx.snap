// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Input component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="viking-input-wrapper"
  >
    <input
      class="viking-input-inner"
      placeholder="test-input"
    />
  </div>
</DocumentFragment>
`;
